src/Opdex.Platform.Infrastructure/Clients/SignalR/PlatformHub.cs

```diff
@@ -1,26 +1,34 @@
 using Microsoft.AspNetCore.SignalR;
+using Opdex.Platform.Common.Configurations;
 using Opdex.Platform.Common.Encryption;
 using Opdex.Platform.Common.Extensions;
+using SSAS.NET;
 using System;
+using System.IO;
 
 namespace Opdex.Platform.Infrastructure.Clients.SignalR;
 
 public class PlatformHub : Hub<IPlatformClient>
 {
     private readonly ITwoWayEncryptionProvider _twoWayEncryptionProvider;
+    private readonly string _authCallback;
 
-    public PlatformHub(ITwoWayEncryptionProvider twoWayEncryptionProvider)
+    public PlatformHub(ITwoWayEncryptionProvider twoWayEncryptionProvider, AuthConfiguration authConfiguration, OpdexConfiguration opdexConfiguration)
     {
         _twoWayEncryptionProvider = twoWayEncryptionProvider ?? throw new ArgumentNullException(nameof(twoWayEncryptionProvider));
+        var authCallbackUri = new Uri(Path.Combine(opdexConfiguration.ApiUrl, authConfiguration.StratisSignatureAuth.CallbackPath));
+        _authCallback = $"{authCallbackUri.Authority}{authCallbackUri.AbsolutePath}";
     }
 
     /// <summary>
-    /// Returns the connection id as an encrypted string.
+    /// Returns a Stratis ID that can be used for authentication.
     /// </summary>
-    /// <returns>Encrypted connection id.</returns>
-    public string GetEncryptedConnectionId()
+    /// <returns>Stratis ID.</returns>
+    public string GetStratisId()
     {
         var encryptedConnectionId = _twoWayEncryptionProvider.Encrypt(Context.ConnectionId);
-        return Base64Extensions.UrlSafeBase64Encode(encryptedConnectionId);
+        var uid = Base64Extensions.UrlSafeBase64Encode(encryptedConnectionId);
+        var stratisId = new StratisId(_authCallback, uid, DateTimeOffset.UtcNow.AddMinutes(5).ToUnixTimeSeconds());
```

Comment:
Will this prefix with `sid:`?

Reply:
`stratisId.ToString()` builds the full URI with the prefix

Reply:
perfect